classes/services/PKPStatsPublicationService.php

```diff
@@ -230,12 +230,17 @@ public function getDefaultArgs(): array
         ];
     }
 
+    /**
+     * Consider/add application specific QB filters
+     */
+    abstract protected function getAppSpecificFilters(array $args = [], StatsPublicationQueryBuilder &$statsQB): void;
```

Comment:
This is another one where you could probably implement it in the base class but do nothing. Then you don't need to implement it in each app class. Only the apps where you want to change something.

Reply:
I have chosen your other suggestion to implement it in the pkp-lib, as you suggested above...